src/detect-engine-hsbd.c

```diff
@@ -268,14 +296,24 @@ int DetectEngineInspectHttpServerBody(ThreadVars *tv,
         Flow *f, uint8_t flags, void *alstate, void *tx, uint64_t tx_id)
 {
     HtpState *htp_state = (HtpState *)alstate;
+    int hsbd_index = 0;
     uint32_t buffer_len = 0;
     uint32_t stream_start_offset = 0;
     const uint8_t *buffer = DetectEngineHSBDGetBufferForTX(tx, tx_id,
                                                      de_ctx, det_ctx,
                                                      f, htp_state,
                                                      flags,
+                                                     &hsbd_index,
                                                      &buffer_len,
                                                      &stream_start_offset);
+
+    if (htp_state->cfg->swf_decompression_enabled) {
+        if (det_ctx->hsbd[hsbd_index].decompressed_buffer_len != 0) {
```

Comment:
scan-build says:
<pre>
detect-engine-hsbd.c:311:13: warning: Dereference of null pointer
        if (det_ctx->hsbd[hsbd_index].decompressed_buffer_len != 0) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
</pre>

Reply:
I've been trying to reproduce the issue without success:
```
./autogen.sh
CC=clang ./configure
scan-build make -j16
cc1: error: -Werror=unused-command-line-argument: no option -Wunused-command-line-argument
```
even it is supposed to be disabled by CLANG_FLAGS in `configure` script:
```
CLANG_CFLAGS="-Wextra -Werror-implicit-function-declaration -Wno-error=unused-command-line-argument"
```

Reply:
Ah I only see it with Scan-build 5.0, maybe you have an older version?

Reply:
With clang-5.0:
```
CC=clang-5.0 ./configure 
scan-builld-5.0 make -j16
```
I get the same result:
```cc1: error: -Werror=unused-command-line-argument: no option -Wunused-command-line-argument```

Could be possible that I'm missing some CFLAGS?

Reply:
Did you try running configure like this: "scan-build-5.0 ./configure". Thats how you're supposed to run it iirc.

Reply:
Ok, it seems that the correct sequence is:
```
CC=clang-5.0 scan-build-5.0 ./configure
scan-build-5.0 make -j16
...
detect-engine-hsbd.c:311:13: warning: Dereference of null pointer
        if (det_ctx->hsbd[hsbd_index].decompressed_buffer_len != 0) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```
Thanks.